recipes/devel/git.yaml

```diff
@@ -28,6 +28,17 @@ buildScript: |
     export CFLAGS="-I${BOB_DEP_PATHS[libs::zlib-dev]}/usr/include -I${BOB_DEP_PATHS[libs::openssl-dev]}/usr/include"
     export LDFLAGS="-L${BOB_DEP_PATHS[libs::zlib-dev]}/usr/lib -L${BOB_DEP_PATHS[libs::openssl-dev]}/usr/lib -Wl,-rpath-link=${BOB_DEP_PATHS[libs::zlib-dev]}/usr/lib -Wl,-rpath-link=${BOB_DEP_PATHS[libs::openssl-dev]}/usr/lib"
 
+    BUILD_ARCH="$(uname -m)"
+    if [[ ${BUILD_ARCH} != ${AUTOCONF_HOST%%-*} ]]; then
```

Comment:
Cross compiling is not just about the architecture. You could also cross compile for the same architecture but e.g. for a different libc. That's what calling `uname -m` and comparing it is not sufficient. The `AUTOCONF` strings are only loosely defined but if `$AUTOCONF_BUILD != $AUTOCONF_HOST` then you are cross compiling...

Reply:
> I'm wondering why this class is not used in the first place...

Just to clarify: by this I meant that the git recipe replicates the setup of CFLAGS and LDFLAGS. But I don't remember why the recipe has not been adapted when the cpackage class was introduced. May be just an oversight...

Reply:
> Cross compiling is not just about the architecture.

Obviously. But you have a point in that the configure script is possibly not smart enough to use the `HOST` dependencies when evaluating the `HOST` capabilities but will instead use the `BUILD` libraries. I've remove the whole cross compilation stuff, because it needs more work if it's used nearly always and I have currently no use-case for it. The values partially depend on which libc is in use and so on, so I'd need more time and - most importantly - more testing to come to a good solution here.

> May be just an oversight...

I've added the `cpackage` class and removed the `LDFLAGS`/`CFLAGS` stuff.

Reply:
I'm not sure I can follow. We practically always cross-compile...